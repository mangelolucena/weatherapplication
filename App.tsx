import React, { Component } from 'react';
import RootStack from './src/navigation/index';
import { Provider } from 'react-redux';
import store from './src/redux/store';


export default class AppComponent extends Component {

  public render() {
    return (
      <Provider store={store}>
        <RootStack />
      </Provider>
    );
  }
}
