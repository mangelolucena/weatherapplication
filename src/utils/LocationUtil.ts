import GetLocation from 'react-native-get-location';

export const getCurrentLocation = async () => {
    const location = new Promise(async (resolve, reject) => {
        GetLocation.getCurrentPosition({
            enableHighAccuracy: true,
            timeout: 15000,
        }).then(location => {
            resolve(location);
        }).catch(error => {
            reject(error)
        })
    });
    return location;
}