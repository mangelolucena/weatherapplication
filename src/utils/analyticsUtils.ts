export function track(name:string){
    doSomeFirebaseLogic(name)
}