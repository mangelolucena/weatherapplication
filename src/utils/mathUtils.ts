import {track} from './analyticsUtils'

export function addNumbers(x:number, y:number) {
    track('addNumbers')
    return x + y;
}