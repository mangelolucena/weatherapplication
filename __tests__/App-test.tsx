/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer, { act } from 'react-test-renderer';


// ui test
// jest.useFakeTimers()
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
test('renders App Screen', async () => {
  await act(async () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});

// unit test sample
import { addNumbers } from '../src/utils/mathUtils';
import { track } from '../src/utils/analyticsUtils'

jest.mock('../src/utils/analyticsUtils');
test('something', () => {
  expect('something')
})

test('add numbers', () => {
  expect(addNumbers(1, 2)).toEqual(3);
  expect(track).toBeCalledWith('addNumbers');
})
