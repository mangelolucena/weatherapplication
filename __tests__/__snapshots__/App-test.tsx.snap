// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App Screen 1`] = `null`;
